evap/staff/importers/enrollment.py

```diff
@@ -341,7 +341,7 @@ def get_merge_hindrances(course_data: CourseData, merge_candidate: Course) -> Li
 
         if len(merge_candidate.evaluations.all()) != 1:
             hindrances.append(_("the existing course does not have exactly one evaluation"))
-        elif merge_candidate.evaluations.get().wait_for_grade_upload_before_publishing != course_data.is_graded:
+        elif merge_candidate.evaluations.all()[0].wait_for_grade_upload_before_publishing != course_data.is_graded:
```

Comment:
hm, I guess I understand why this works, but I think the intent of sparing the extra query gets across better if we move the ~~list~~ queryset into its own variable.

While we are at the topic of performance, we _could_ [limit the number of evaluations](https://docs.djangoproject.com/en/4.1/topics/db/queries/#limiting-querysets) returned here as we only need 2 to prove that there are too many, but in our domain there shouldn't be more than, say, 10 evaluations anyways, so probably not worth it.

---

When we get Python 3.10, it will be even nicer:
```python
match list(merge_candidate.evaluations.all()):
    case [existing_evaluation]:
        if existing_evaluation.wait_for_grade_upload_before_publishing != course_data.is_graded:
            hindrances.append(_("the evaluation of the existing course has a mismatching grading specification"))
    case _:
        hindrances.append(_("the existing course does not have exactly one evaluation"))
```

Reply:
Moved into separate variable.

Limited prefetching is a good idea, but it's also a slippery slope. I wouldn't want to prefetch the limited set into `evaluations`, because other code might (rightfully) assume that it has all evaluations in that attribute. So we'd need a separate attribute, but then we'd have `evaluations` not prefetched for other accesses.
As you said, I don't think there's much to gain here, so I'll leave that for now.